R/QFeatures-longForm.R

```diff
@@ -0,0 +1,139 @@
+##' @rdname QFeatures-class
+##'
+##' @param colvars A `character()` that selects column(s) in the `colData`.
+##'
+##' @param rowvars A `character()` with the names of the `rowData`
+##'     variables (columns) to retain in any assay.
+##'
+##' @param index When object is an instance of class `QFeatures`, a `numeric(1)`
+##'     indicating what assay within each `SummarizedExperiment` object to
+##'     return. Default is `1L`. If object is a `SummarizedExperiment`, a
+##'     `numeric()` indicating what assays to pull and convert. Default is to
+##'     use all assays.
+##'
+##' @importFrom MultiAssayExperiment longForm
+##' @importFrom reshape2 melt
+##'
+##' @importFrom BiocGenerics longForm
+##'
+##' @exportMethod longForm
+##'
+##' @aliases longForm longForm,QFeatures
+##' @aliases longFormat
+setMethod("longForm", "QFeatures",
+          function(object, colvars = NULL,
+                   rowvars = NULL,
+                   index = 1L)
+              longFormQFeatures(object, colvars, rowvars, index))
+
+##' @rdname QFeatures-class
+##'
+##' @exportMethod longForm
+##'
+##' @aliases longForm,SummarizedExperiment
+setMethod("longForm", "SummarizedExperiment",
+          function(object, colvars = NULL,
+                   rowvars = NULL,
+                   index = seq_along(assays(object)))
+              longFormSE(object, colvars, rowvars, index, na.rm = FALSE))
+
+longFormQFeatures <- function(object, colvars = NULL,
```

Comment:
The longFormSE allows to remove missing values (`na.rm` argument). Is it intentional that `longFormQFeatures` doesn't?

Also both function take an `index` argument. Should this not be `i` to comply with the interface of the other functions/methods in the package?

I would personnally prefer a harmonized interface.